probably/pgcl/parser.py

```diff
@@ -237,23 +295,33 @@ def _parse_literal(t: Tree) -> Expr:
         return BoolLitExpr(False)
     elif t.data == 'nat':
         return NatLitExpr(int(_child_str(t, 0)))
-    elif t.data == 'float':
-        return FloatLitExpr(Decimal(_child_str(t, 0)))
+    elif t.data == 'real':
+        return RealLitExpr(Decimal(_child_str(t, 0)))
     elif t.data == 'infinity':
-        return FloatLitExpr.infinity()
+        return RealLitExpr.infinity()
     else:
         raise Exception(f'invalid AST: {t.data}')
 
 
+def _parse_distribution(t: Tree) -> Expr:
+    assert t.data in distributions
+    param_count, constructor = distributions[t.data]
+    params = []
+    for i in range(param_count):
+        param = _parse_expr(_child_tree(t, i))
+        if has_variable(param):
+            raise SyntaxError(
+                "In distribution parameter expressions, no variables are allowed. - Forgot parameter declaration?")
```

Comment:
I think this check shouldn't happen in the parser since it is specific to backend support (prodigy).

Reply:
I agree with Philipp. We should allow this and either accept or reject expressions that contain variables in the tools working with the AST provided by probably.